plugins/stats_pusher_statsd/plugin.c

```diff
@@ -88,7 +96,10 @@ static void stats_pusher_statsd(struct uwsgi_stats_pusher_instance *uspi, time_t
 	while(um) {
 		uwsgi_rlock(uwsgi.metrics_lock);
 		// ignore return value
-		if (um->type == UWSGI_METRIC_GAUGE) {
+		if (u_stats_pusher_statsd.no_workers && strstr(um->name, "worker.") != NULL) {
```

Comment:
place this check before the lock, and use a "goto next" to the end of the while block (where um is redefined).

Instead of strstr use uwsgi_starts_with(src, src_len, dst, dst_len) that is faster and safer